distributed/active_memory_manager.py

```diff
@@ -46,7 +44,7 @@ class ActiveMemoryManagerExtension:
     """
 
     #: Back-reference to the scheduler holding this extension
-    scheduler: Scheduler
+    scheduler: scheduler_module.Scheduler
```

Comment:
we have to rename `scheduler` to `scheduler_module` to avoid clashing with the `scheduler` field here